### PR TITLE
feat: support unbinding bound text

### PR DESCRIPTION
**Pull request by ad1992**
fixes https://github.com/excalidraw/excalidraw/issues/4591
Excalidraw (18)

- [x] Add spec

### CODE REVIEW

File: src/actions/actionUnbindText.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  contextItemLabel: "labels.unbindText",
+  perform: (elements, appState) => {
+    const selectedElements = getSelectedElements(
+      getNonDeletedElements(elements),
+      appState,
+    );
+    selectedElements.forEach((element) => {
+      const boundTextElement = getBoundTextElement(element);
+      if (boundTextElement) {
+        mutateElement(boundTextElement as ExcalidrawTextElement, {
```
Comment (dwelle) [inline src/actions/actionUnbindText.tsx:22]: this type assertion (and the one below) should be unnecessary

File: src/actions/actionUnbindText.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          containerId: null,
+        });
+        mutateElement(element, {
+          boundElements: element.boundElements?.filter(
+            (ele) =>
+              !isTextElement(
+                ele as ExcalidrawTextElement | ExcalidrawLinearElement,
+              ),
```
Comment (dwelle) [inline src/actions/actionUnbindText.tsx:30]: to make this future-proof, we should only remove the target element using the `boundTextElement.id`

**Comment by dwelle** [comment]
Maybe put the contextmenu item next to the group? I think putting it next to duplicate/delete items isn't ideal.

image

**Comment by dwelle** [comment]
Also, I think we should re-measure the text after unbound so that it ends up with the bounding box you get once you edit the text in the future.

One last thing: what about also supporting it the other way round — when select a text element and a container, support `bind text to container`? Maybe for another PR.

**Comment by dwelle** [comment]
> should we have an action in sidebar ?

I would not, for now. Let's conserve UI buttons. We can decide on that later based on feedback.

**Comment by ad1992** [comment]
> Also, I think we should re-measure the text after unbound so that it ends up with the bounding box you get once you edit the text in the future.

done
> 
> One last thing: what about also supporting it the other way round — when select a text element and a container, support `bind text to container`? Maybe for another PR.

Yep lets keep for anothe PR

File: src/actions/actionUnbindText.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    const selectedElements = getSelectedElements(
+      getNonDeletedElements(elements),
+      appState,
+    );
+    selectedElements.forEach((element) => {
+      const boundTextElement = getBoundTextElement(element);
+      if (boundTextElement) {
+        mutateElement(boundTextElement as ExcalidrawTextElement, {
+          containerId: null,
+          originalText: boundTextElement.text,
```
Comment (dwelle) [inline src/actions/actionUnbindText.tsx:21]: this will persist the soft & hard line breaks — not sure that's what we wanna do

Comment (ad1992) [reply src/actions/actionUnbindText.tsx:21]: what do you mean ? we want to make sure the text which was last edited is exactly same as original text once unbind

Comment (ad1992) [reply src/actions/actionUnbindText.tsx:21]: Hmm ok I get it, you mean treplace with original text but with recomputed measurement

Comment (ad1992) [reply src/actions/actionUnbindText.tsx]: done 👇🏻 
Excalidraw (19)

Comment (dwelle) [reply src/actions/actionUnbindText.tsx]: 👍 btw, once we support #4267 we can keep the bounded dimensions

**Review by dwelle — APPROVED** [review]

File: src/actions/actionUnbindText.tsx (hunk 3 above)
Comment (ad1992) [reply src/actions/actionUnbindText.tsx]: yep

**Pull request merged**